<launch>
    <!-- <node type="rviz" name="rviz" pkg="rviz" args="-d $(find package_name)/rviz/multi_teleop.rviz" /> -->
    <arg name="own_name" default="car38" />
    
    <!-- Total number of cars (including self) -->
    <arg name="car_count" default = "1"/>
    
    <!-- Other car names (if car_count is 2, then nodes will read own_name, other_car1, then disregard the rest) -->
    <arg name="other_car1" default = "car2"/>
    <arg name="other_car2" default = "car3"/>
    <arg name="other_car3" default = "car4"/>
    <arg name="other_car4" default = "car5"/>
    <arg name="other_car5" default = "car6"/>
    <arg name="other_car6" default = "car7"/>
    <arg name="other_car7" default = "car8"/>

    <!-- Set to 1 if you want to run the map_server -->
    <arg name="map_server" value = "1"/>

    <!-- Launch  map server-->
    <group if="$(arg map_server)">
       <!--  <include file="$(find mushr_sim)/launch/map_server.launch" /> -->
	<include file="$(find mushr_rhc_ros)/launch/map_server.launch" />
    </group>

    <!-- Launch Car Node -->
    <group ns="$(arg car_name)">
        <node pkg="nhttc_ros" type="nhttc_ros_node" name="controller" output="screen">
            <param name="car_count" value="$(arg car_count)"/>
            <param name="own_name" value="$(arg own_name)"/>
            <param name="other_car1" value="$(arg other_car1)"/>
            <param name="other_car2" value="$(arg other_car2)"/>
            <param name="other_car3" value="$(arg other_car3)"/>
            <param name="other_car4" value="$(arg other_car4)"/>
            <param name="other_car5" value="$(arg other_car5)"/>
            <param name="other_car6" value="$(arg other_car6)"/>
            <param name="other_car7" value="$(arg other_car7)"/>
            
            <param name="solver_time" value="20"/>
            <param name="sim" value="0"/>
            
            <!-- Currently set this equal to car count but this may need to be fixed -->
            <param name="max_agents" value="$(arg car_count)"/>
        </node>
    </group>

    <node pkg="nhttc_ros" type="nhttc_pose_goal_init.py" name="initializer" output="screen">
        <param name="car_count" value="$(arg car_count)"/>
        <!-- Each arg is a tuple of (car_name,boolean) of whether it is a sim car or not -->
        <param name="own_name-sim" value="$(arg own_name),0"/>
        <param name="other_car1-sim" value="$(arg other_car1),1"/>
        <param name="other_car2-sim" value="$(arg other_car2),1"/>
        <param name="other_car3-sim" value="$(arg other_car3),1"/>
        <param name="other_car4-sim" value="$(arg other_car4),1"/>
        <param name="other_car5-sim" value="$(arg other_car5),1"/>
        <param name="other_car6-sim" value="$(arg other_car6),1"/>
        <param name="other_car7-sim" value="$(arg other_car7),1"/>
    </node>
    <!-- <node pkg="nhttc_ros" type="nhttc_publisher.py" name="waypointer"/> -->
    <!-- Launch third car, fourth car, etc by following above template -->

</launch>
